### ISSUE TITLE
mklittlefs crashes on symlink 

### ISSUE THREAD

**Issue by mhaberler** [earlephilhower/mklittlefs#22]
try this (macOS but likely others as well:

````
$ ln -s . internal
$ mklittlefs . -c /tmp/image.bin

/js/jquery/jquery.terminal-2.29.5.min.js.gz
lfs warn:475: No more free space 20
lfs_write error(-28): File system is full.

error adding file!
Error for adding content from jquery!
....
````
I think the best fix would be to (optionally) skip symbolic links

**Comment by earlephilhower** [comment]
Good catch, that was definitely not something we'd thought about before.  Replacing `stat` with `lstat` (on non-WIN32 systems) and checking flags should handle things.

Would you like to propose a PR to fix, since you found it and have a solution?

**Comment by mhaberler** [comment]
will give it a try

**Comment by earlephilhower** [comment]
Great!

As I was thinking about it, it may be a little more complicated.  You'll want to do `stat` to see if it's a directory, and then if do `lstat` to see if it's a symlink to a directory and if so skip.  But you don't want to skill all symlinks (i.e. a linked file), I imagine, just directories.

**Comment by earlephilhower** [comment]
Fixed by #23
